iroha/src/bridge.rs

```diff
@@ -695,6 +754,32 @@ pub mod query {
             })
     }
 
+    /// A helper function for decoding information about external assets from the query result.
+    ///
+    /// Each `ExternalAsset` is encoded and stored in the bridge external assets asset
+    /// (`bridge_external_assets_asset_definition_id`) store and indexed by a name of the asset.
+    /// The given query result may not contain the above values, so this function can fail,
+    /// returning `None`.
+    pub fn decode_external_assets<'a>(
+        query_result: &'a QueryResult,
+    ) -> Option<impl Iterator<Item = ExternalAsset> + 'a> {
```

Comment:
Maybe use TryFrom\TryInto traits instead?

Reply:
@humb1t can you clarify what do you mean? Something like `impl From<&QueryResult> for Iterator<Item = ExternalAsset>`?

Reply:
@vmarkushin yes, we use rust std traits when we can.

Reply:
@humb1t here is comparison of 2 variants
```rust
impl<'a> TryFrom<&'a QueryResult> for Box<dyn Iterator<Item = ExternalAsset> + 'a> {
...
}

fn foo() {
    let assets = Box<dyn Iterator<Item = ExternalAsset>>::try_from(&query_result);
    assets.for_each(|asset| ...)
}
```
vs
```rust
fn foo() {
    let assets = decode_external_assets(&query_result);
    assets.for_each(|asset| ...)
}
```
In my opinion, the first variant looks noisy. What do you think?

Reply:
I agree - can you stay with `fn` variant but create a tech debt issue on Hyperledger Jira to work on this code later?

Reply:
@humb1t sure. I plan to migrate all the query_* and decode_* functions into one `Bridge` structure to make it easy for requesting bridge info.